### ISSUE TITLE
Allow manual configuration of the path of the GeoIP2 database

### ISSUE THREAD

**Issue by jsabater** [piwik/piwik#22557]
When you already have the geolocation available in the system, or when you configure Matomo with `enable_internet_features = 0`, it would be useful to be able to configure Matomo (`Settings: System: Geolocation`) with the path to the database file. Matomo would, then, not worry about updates or anything, but just use it. 

An example scenario would be:

* Using Debian with the `geoipupdate` package, which has a cron job to update MaxMind's database periodically.
* Disabling Matomo's internet access because it's running in an isolated environment, behind a reverse proxy.

In this scenario, when accessing the `Settings: System: Geolocation` menu entry, when selecting the "DBIP / GeoIP 2 (Php)" option, one would be offered different options at the bottom "Setup automatic updates of geolocation databases":

* Let Matomo handle updates (with the current options, or similar ones).
* Provide a path to the database, which will be updated by the system.

When using the second option, just a field with the path would be necessary:

Path to GeoIP2 database: /var/lib/GeoIP/GeoLite2-City.mmdb

I understand that this would support a DB-IP database as well.

Thanks.

**Comment by randy-innocraft** [comment]
Hi @jsabater,
Thank you for bringing this to our attention and for your valuable input.
Your suggestion seems like a valuable enhancement to our product.
We will forward this to our Product team for review and future consideration.
If you have any additional details or questions, please feel free to share them here.

**Comment by sgiehl** [comment]
@jsabater This should already be kind of possible if you adjust the DI config. If there is not yet a `config.php` in the config directory you can create one with the following content:

```php
<?php

return [
    'path.geoip2' => '/var/lib/GeoIP/'
];
```

The geoip files still need to have an expected name to be picked up automatically:
https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/plugins/GeoIp2/LocationProvider/GeoIp2.php#L41-L48

**Comment by jsabater** [comment]
Hi, @sgiehl , and thanks for the indications. I think that you mean the `plugins/GeoIp2/config/config.php` file, which contains:

```php
<?php

return [
    'path.geoip2' => Piwik\DI::string('{path.root}/misc/'),
    'geopip2.ispEnabled' => true
];
```

From your words, I understand I would change it into something like this, correct?

```php
<?php

return [
    'path.geoip2' => '/var/lib/GeoIP/',
    'geopip2.ispEnabled' => true
];
```

I could add a task to my Ansible playbook to do this substitution as part of the application upgrade.

I take it that would be all? Would there be any way I could programmatically enable the Geolocation plug-in at `Settings: System: Geolocation` to use the `DBIP / GeoIP 2 (Php)` option?

**Comment by sgiehl** [comment]
No. I really meant creating a config file in config directory. That file will automatically overwrite the config in geoip plugin.
Once a geoip file is present you should be able to activate the provider in settings.

**Comment by jsabater** [comment]
@sgiehl , thanks for the clarification. I'll create a `config/config.php` file with the content mentioned above. From your words, I take that I will have to enable the right Geolocation plug-in using the web interface, meaning there is no programmatical way to do so?

**Comment by karolyi** [comment]
@sgiehl can you please specify the _exact path_ of that `config.php` that will override the one in `plugins/GeoIp2/config/`?

**Comment by sgiehl** [comment]
@karolyi As mentioned above that is `config/config.php`. That file might not exist by default.

**Comment by karolyi** [comment]
Thanks. I already solved it by symlinking `GeoLite2-City.mmdb` under `misc/`. Took a little research but works out of the box, without having to add any extra configuration.
